Solutions/Azure Active Directory/Analytic Rules/NRT_PIMElevationRequestRejected.yaml

```diff
@@ -18,24 +18,38 @@ tags:
 query: |
   AuditLogs
   | where ActivityDisplayName =~'Add member to role completed (PIM activation)'
-  | where Result == "failure"
-  | extend Role = tostring(TargetResources[3].displayName)
-  | extend User = tostring(TargetResources[2].displayName)
+  | where Result =~ "failure"
+  | mv-apply ResourceItem = TargetResources on 
+    (
+        where ResourceItem.type == "Role"
+        | extend Role = trim(@'"',tostring(ResourceItem.displayName))
+    )
+  | mv-apply ResourceItem = TargetResources on 
+    (
+        where ResourceItem.type == "User"
+        | extend User = trim(@'"',tostring(ResourceItem.userPrincipalName))
+    )
   | project-reorder TimeGenerated, User, Role, OperationName, Result, ResultDescription
-  | extend InitiatingUser = tostring(parse_json(tostring(InitiatedBy.user)).userPrincipalName)
-  | extend IPCustomEntity = tostring(parse_json(tostring(InitiatedBy.user)).ipAddress)
+  | where isnotempty(InitiatedBy.user)
```

Comment:
@petebryan Is it within the logic to remove instances where the operation is performed by app?